src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdateOpResolver.scala

```diff
@@ -71,6 +74,16 @@ private[marathon] class InstanceUpdateOpResolver(
     }
   }
 
+  private[this] def updateExistingInstanceSafe(id: Instance.Id)(applyOperation: Instance => InstanceUpdateEffect)(implicit ec: ExecutionContext): Future[InstanceUpdateEffect] = {
```

Comment:
Instead of replicating `updateExistingInstance` maybe we should extend it and add an additional boolean parameter (with default) to indicate wheter we should fail or not?